airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/AbstractDbSource.java

```diff
@@ -180,6 +183,35 @@ public AutoCloseableIterator<AirbyteMessage> read(final JsonNode config,
         });
   }
 
+  private void validateSourceSchema(Map<String, TableInfo<CommonField<DataType>>> fullyQualifiedTableNameToInfo,
+                                    ConfiguredAirbyteCatalog catalog,
+                                    Database database) {
+    for (final ConfiguredAirbyteStream airbyteStream : catalog.getStreams()) {
+      final AirbyteStream stream = airbyteStream.getStream();
+      final String fullyQualifiedTableName = getFullyQualifiedTableName(stream.getNamespace(),
+          stream.getName());
+      if (!fullyQualifiedTableNameToInfo.containsKey(fullyQualifiedTableName)) {
+        continue;
+      }
+      final TableInfo<CommonField<DataType>> table = fullyQualifiedTableNameToInfo.get(fullyQualifiedTableName);
+      final List<Field> fields = table.getFields()
+          .stream()
+          .map(this::toField)
+          .distinct()
+          .collect(Collectors.toList());
+      final JsonNode currentJsonSchema = fieldsToJsonSchema(fields);
+
+      final JsonNode catalogSchema = stream.getJsonSchema();
+      if (!catalogSchema.equals(currentJsonSchema)) {
```

Comment:
Does this account for column selection? The user could have specifically decided to not include certain columns in the configured catalog

Reply:
> Does this account for column selection? The user could have specifically decided to not include certain columns in the configured catalog

it is another one excellent example of why throwing an exception here is a bad idea
Column selection - is a valid case if the actual schema and catalog schema are not the same and sync will be successful

This PR is not a fix, it's just additional logging to detect quickly similar issues

Reply:
+1 on not throwing an exception.

Reply:
I think the confusion arises here since you're conflating JsonSchema (Airbyte type) vs source schema. 

I think that we should log changes in the Airbyte type. So, you'd need to change the logic to grab fields that are only present in the ``catalogSchema``. 

At the end of the day, normalization only cares about the Airbyte type (and not the source schema). Let me know if that makes sense - happy to jump on a call to explain what I'm thinking.

Reply:
> I think the confusion arises here since you're conflating JsonSchema (Airbyte type) vs source schema.
> 
> I think that we should log changes in the Airbyte type. So, you'd need to change the logic to grab fields that are only present in the `catalogSchema`.
> 
> At the end of the day, normalization only cares about the Airbyte type (and not the source schema). Let me know if that makes sense - happy to jump on a call to explain what I'm thinking.

@akashkulk the example that you give with normalization and airbyte_type is just one of the possible cases in which the sync can fail if the user has changed the database structure
I have described all potential fails [here](https://github.com/airbytehq/airbyte/pull/21844#discussion_r1093321662)
Therefore, in my opinion, it is better to leave the logging as it is.